layer-atlas/src/main/java/com/layer/atlas/util/imagepopup/AtlasImagePopupActivity.java

```diff
@@ -103,6 +129,65 @@ public static void init(LayerClient layerClient) {
         MessagePartRegionDecoder.init(layerClient);
     }
 
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.atlas_image_menu, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+       if(item.getItemId() == R.id.action_save) {
+           if(checkPermission()) {
+               saveImageToGallery();
+           } else {
+               requestPermission();
+           }
+           return true;
+       }
+       return super.onOptionsItemSelected(item);
+    }
+
+    private void saveImageToGallery() {
+        try {
```

Comment:
The whole method has to be in try catch? Only decode method can throw exception?